catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchSource.java

```diff
@@ -936,4 +950,19 @@ private void applyFilters(OpenSearchFilterVisitorObject visitor, WebClient clien
             client.replaceQueryParam(URL_SRC_PARAMETER, "");
         }
     }
+
+    private List<Metacard> processAdditionalForeignMarkups(Element element, String id) {
+        List<Metacard> metacards = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(markUpList)) {
+            XMLOutputter xmlOutputter = new XMLOutputter();
+            for (String markup : markUpList) {
+                if (element.getName().equals(markup)) {
```

Comment:
Is it possible for an element to have more than one name? Or are we going to have multiple elements with that will match a given markup?
If it's the latter, we might want to coerce the injected list into a HashSet, and use a lookup instead of iterating over the whole list each time.
```
if (markUpSet.contains(element.getName) {
  metacards.add(parseContent(xmlOutputter.outputString(element), id);
}
```